extmod/machine_signal.c

```diff
@@ -58,7 +58,7 @@ STATIC mp_obj_t signal_make_new(const mp_obj_type_t *type, size_t n_args, size_t
         // If first argument isn't a Pin-like object, we filter out "invert"
         // from keyword arguments and pass them all to the exported Pin
         // constructor to create one.
-        mp_obj_t pin_args[n_args + n_kw * 2];
+        mp_scoped_alloc(mp_obj_t, pin_args, n_args + n_kw * 2);
```

Comment:
It is unfortunate, but the only way it will work for VLA (syntax of removed code), alloca() and heap allocation.